backend/src/revisions/revisions.service.ts

```diff
@@ -219,4 +226,73 @@ export class RevisionsService {
       await this.revisionRepository.save(revision);
     }
   }
+
+  // Delete all old revisions everyday on 0:00 AM
+  @Cron('0 0 * * *')
+  async handleCron(): Promise<void> {
+    return await this.removeOldRevisions();
+  }
+
+  // Delete all old revisions 5 sec after startup
+  @Timeout(5000)
+  async handleTimeout(): Promise<void> {
+    return await this.removeOldRevisions();
+  }
+
+  async removeOldRevisions(): Promise<void> {
+    const currentTime = new Date().getTime();
+    const revisionRetentionDays: number = this.noteConfig.revisionRetentionDays;
+    if (revisionRetentionDays <= 0) {
+      return;
+    }
+
+    const notes: Note[] = await this.noteRepository.find();
+    for (const note of notes) {
+      const revisions: Revision[] = await this.revisionRepository.find({
+        where: {
+          note: { id: note.id },
+        },
+        order: {
+          createdAt: "ASC",
+        },
+      });
+
+      const oldRevisions = revisions
+        .slice(0, -1) // always keep the latest revision
+        .filter(
+          (revision) =>
+            new Date(revision.createdAt).getTime() <=
+            currentTime - revisionRetentionDays * 24 * 60 * 60 * 1000,
```

Comment:
To improve performance, this second value could be calculated outside the loop, so before the `.filter` method call.